Steer/DigitizerWorkflow/src/TPCDigitRootWriterSpec.cxx

```diff
@@ -0,0 +1,209 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   TPCDigitRootFileWriterSpec.cxx
+/// @author Matthias Richter, Sandro Wenzel
+/// @since  2018-04-19
+/// @brief  Processor spec for a ROOT file writer for TPC digits
+
+#include "TPCDigitRootWriterSpec.h"
+#include "Framework/CallbackService.h"
+#include "Framework/ControlService.h"
+#include "TPCBase/Sector.h"
+#include "TPCBase/Digit.h"
+#include <SimulationDataFormat/MCCompLabel.h>
+#include <SimulationDataFormat/MCTruthContainer.h>
+#include <TFile.h>
+#include <TTree.h>
+#include <TBranch.h>
+#include <memory> // for make_shared, make_unique, unique_ptr
+#include <stdexcept>
+#include <sstream>
+#include <string>
+#include <vector>
+
+using namespace o2::framework;
+using SubSpecificationType = o2::framework::DataAllocator::SubSpecificationType;
+
+namespace o2
+{
+namespace TPC
+{
+
+template <typename T>
+TBranch* getOrMakeBranch(TTree& tree, const char* basename, int sector, T* ptr)
+{
+  std::stringstream stream;
+  stream << basename << "_" << sector;
+  const auto brname = stream.str().c_str();
+  if (auto br = tree.GetBranch(brname)) {
+    br->SetAddress((void*)&ptr);
```

Comment:
such casts should better be avoided, I know it's a stupid ROOT limitation not to allow a `const` pointer. Better use `const_cast` and add a comment